pandas/io/formats/xml.py

```diff
@@ -247,17 +251,37 @@ def other_namespaces(self) -> dict:
 
         return nmsp_dict
 
-    def build_attribs(self) -> None:
+    def build_attribs(self, d: dict[str, Any], elem_row: Any) -> None:
```

Comment:
don't luv the modification inline here, can you copy and return a new object?

Reply:
We could return it like that

Reply:
great, yeah i think that's a nicer pattern

Reply:
Yep definitively